### ISSUE TITLE
asymptote/base /map.asy does not compile

### ISSUE THREAD

**Issue by Harniver** [vectorgraphics/asymptote#470]
The following minimal asy file:
```
import map;
```
Does not compile, producing the error:
```
  type(key,"Key");
  ^
/usr/local/Cellar/asymptote/2.90/share/asymptote/map.asy: 5.3: no matching variable 'type'
  type(value,"Value");
  ^
/usr/local/Cellar/asymptote/2.90/share/asymptote/map.asy: 6.3: no matching variable 'type'
  type("map",name);
  ^
/usr/local/Cellar/asymptote/2.90/share/asymptote/map.asy: 38.3: no matching variable 'type'
import map;
       ^
test.asy: 1.8: could not load module 'map'
import map;
^
test.asy: 1.1: no type of name 'map'
import map;
^
test.asy: 1.1: qualifier is not a record
```

This has been tested on a MacBook Pro running macOS 14.1.2, though it doesn't seem the kind of issue that may be system-related.

**Comment by johncbowman** [comment]
Sorry, we somehow forgot to update the map module to use templated imports. This is now fixed and will be included in the upcoming 2.91 release. In the meantime, you can use a local copy of https://raw.githubusercontent.com/vectorgraphics/asymptote/HEAD/base/map.asy
like this:

```
from map(Key=string, Value=int) access map;
map M=map(Default=-1);
M.add("z",2);
M.add("a",3);
M.add("d",4);
write(M.lookup("a"));
write(M.lookup("y"));
```

**Comment by Harniver** [comment]
Awesome, thanks!
